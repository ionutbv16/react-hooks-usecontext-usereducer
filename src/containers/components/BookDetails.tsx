import React from 'react';
import styled from 'styled-components';
import Card from "@material-ui/core/Card";
import { makeStyles } from '@material-ui/core/styles';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import { IBooks } from '../../types';

const useStyles = makeStyles({
  root: {
    maxWidth: 345,
    cursor: "default",
  },
  media: {
    height: 140,
    width: 345,
    cursor: "default",
  },
  content: {
    cursor: "default",
  }
});

const Wrapper = styled.div`
  font-size: 14px;
  flex:2;
  padding: 14px;
  display: flex;
  align-items: center;
`;

const CardDetails: React.FC<IBooks> = ({ id, title, author, cover, isbn }) => {
  
  const classes = useStyles();

  return (
  <Wrapper>
    <Card >
      <CardActionArea>
        <img src={cover} />
        <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
            {title}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
            by {author}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
            Isbn: {isbn}
            </Typography>
          </CardContent>
        </CardActionArea>
        <CardActions>
      </CardActions>
    </Card> 
  </Wrapper>
  )
  };

export default CardDetails;
