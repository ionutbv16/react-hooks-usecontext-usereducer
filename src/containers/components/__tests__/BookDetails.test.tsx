import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import BookDetails from '../BookDetails';

describe('Test BookDetails Component', () => {
    it('should render without error', () => {
    const props = {
        title: 'title',
        author: 'Author',
        cover: 'https://lorempixel.com/640/480/?9d8b21dd8f74',
        id: "89cae71c-fbe5-445c-8299-6de7a88ea5ab",
        isbn: "a898978",
    }
    const component = renderer.create(<MemoryRouter><BookDetails {...props} /></MemoryRouter>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    });
});
