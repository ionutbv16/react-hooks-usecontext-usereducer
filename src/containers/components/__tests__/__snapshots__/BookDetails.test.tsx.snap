// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BookDetails Component should render without error 1`] = `
<div
  className="sc-bdVaJa IMKmx"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        src="https://lorempixel.com/640/480/?9d8b21dd8f74"
      />
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          title
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          by 
          Author
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Isbn: 
          a898978
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    />
  </div>
</div>
`;
